kasa/device.py

```diff
@@ -35,7 +35,10 @@ class WifiNetwork:
 _LOGGER = logging.getLogger(__name__)
 
 
-class Device(ABC):
+T = TypeVar("T", bound="Device")
+
+
+class Device(ABC, Generic[T]):
```

Comment:
I don't think it's necessary to use generics here.  If you use `Mapping` and do this in the base class `__init__`:

`self._children: Mapping[str, "Device"] = {}`

and then in the derived classes `__init__`:

`self._children: Mapping[str, "SmartDevice"] = {}`

and:

`self._children: Mapping[str, "IotDevice"] = {}`

Then mypy is happy and you can go back to `"Device"` everywhere you used the generic type.  Unless I'm missing something.

Reply:
Great, I removed the generics and I think this is now ready to go :+1: